packages/e2e-test-utils-playwright/src/admin/create-new-post.js

```diff
@@ -32,14 +30,16 @@ export async function createNewPost( {
 
 	await this.visitAdminPage( 'post-new.php', query );
 
-	const canvasReadyLocator = legacyCanvas
-		? this.page.locator( '.edit-post-layout' )
-		: this.page
-				.frameLocator( '[name=editor-canvas]' )
-				.locator( 'body > *' )
-				.first();
-
-	await canvasReadyLocator.waitFor();
+	// Wait for both iframed and non-iframed canvas locator and resolve once the
+	// currently available one is ready.
+	await Promise.any( [
+		this.page.locator( '.wp-block-post-content' ).waitFor(),
```

Comment:
Ah, yes - that is a selector that you'd need to pierce the canvas iframe for if the canvas was iframed. Otherwise, it will resolve immediately. I will add a note there.

Reply:
https://github.com/WordPress/gutenberg/pull/53744/commits/5d13020c94b120d2a61723330f24af5764d8cfad